# Import Dependencies

In [53]:
import pandas as pd
import yfinance as yf

pd.options.display.float_format = '{:.6f}'.format

# Data Acquisition

In [54]:
sbux = yf.Ticker("SBUX")
test = sbux.history(period="1mo")

# Test Data

In [56]:
test.reset_index(inplace=True)
test = test.iloc[18:]
test = test[['Date', 'Open']]
test['Date'] = test['Date'].dt.strftime('%Y-%m-%d')

In [58]:
test.reset_index(inplace=True, drop=True)
test.columns = ['date', 'true_value']
test

date  true_value
0  2023-12-20   96.360001
1  2023-12-21   95.080002
2  2023-12-22   94.889999

In [59]:
pred = pd.read_csv(r"C:\Users\nene0\OneDrive\바탕 화면\Python Learning\DataScienceMod2_LFZ\TimeSeries\stock_prediction.csv")
pred.columns = ['prediction']

In [60]:
new_df = test.join(pred)
new_df

date  true_value  prediction
0  2023-12-20   96.360001   96.728549
1  2023-12-21   95.080002   96.782498
2  2023-12-22   94.889999   96.909528

In [61]:
new_df['difference'] = new_df['true_value'] - new_df['prediction']

In [62]:
new_df

date  true_value  prediction  difference
0  2023-12-20   96.360001   96.728549   -0.368548
1  2023-12-21   95.080002   96.782498   -1.702496
2  2023-12-22   94.889999   96.909528   -2.019529

# Summary

Fascinating insights! Initially, the deviation or error in the first prediction, particularly the 12/20/2023 open price, was approximately 0.37, showing a relatively acceptable margin. However, this error notably escalated by the following day.

Creating a flawless or nearly flawless stock price forecast proves incredibly challenging due to the multitude of factors influencing market fluctuations. Presently (as of 12/22/2023), the Starbucks stock price is grappling with several adverse issues, including boycotts and the emergence of a formidable competitor in their key market, China.

Conclusively, while ARIMA models excel in short-term forecasting, they face limitations when dealing with extended forecast periods, especially amidst intricate datasets like stock prices, where numerous variables impact the price dynamics. In such scenarios, employing sophisticated models such as LSTM networks might yield more accurate predictions.